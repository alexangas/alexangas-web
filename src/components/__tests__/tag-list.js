import React from "react"
import renderer from "react-test-renderer"

import TagList from "../tag-list"

describe("TagList", () => {
  it("renders no tags", () => {
    const testRenderer = renderer.create(
      <TagList />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders one tag", () => {
    const testRenderer = renderer.create(
      <TagList tags={['tag1']} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders multiple tags", () => {
    const testRenderer = renderer.create(
      <TagList tags={['tag1', 'tag2', 'tag3']} />
    )
    const tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
