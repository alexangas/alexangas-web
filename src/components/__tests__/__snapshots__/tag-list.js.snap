// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList renders multiple tags 1`] = `
<div
  className="tags is-inline-block-tablet"
>
  <a
    className="tag"
    href="/blog/tags/tag-1/"
  >
    <span
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
        />
      </svg>
    </span>
    tag1
  </a>
  <a
    className="tag"
    href="/blog/tags/tag-2/"
  >
    <span
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
        />
      </svg>
    </span>
    tag2
  </a>
  <a
    className="tag"
    href="/blog/tags/tag-3/"
  >
    <span
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
        />
      </svg>
    </span>
    tag3
  </a>
</div>
`;

exports[`TagList renders no tags 1`] = `null`;

exports[`TagList renders one tag 1`] = `
<div
  className="tags is-inline-block-tablet"
>
  <a
    className="tag"
    href="/blog/tags/tag-1/"
  >
    <span
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
        />
      </svg>
    </span>
    tag1
  </a>
</div>
`;
